shared/src/hooks/useENS.ts

```diff
@@ -0,0 +1,22 @@
+import { ethers } from "ethers";
+import { useEffect, useState } from "react";
+import { useWeb3Context } from "shared/lib/hooks/web3Context";
+
+const useENS = (address: string) => {
+  const { provider } = useWeb3Context();
```

Comment:
The better approach is to do

```js
const {library} = useWeb3React();
```

We only use `useWeb3Context` for the unconnected wallet state